gs_quant/test/timeseries/test_arima.py

```diff
@@ -0,0 +1,192 @@
+                    Timestamp('1989-01-20 00:00:00'): 0.13648992776870728,
+                    Timestamp('1989-01-23 00:00:00'): 0.13573989272117615,
+                    Timestamp('1989-01-24 00:00:00'): 0.13761481642723086,
+                    Timestamp('1989-01-25 00:00:00'): 0.13761481642723086,
+                    Timestamp('1989-01-26 00:00:00'): 0.14061467349529266,
+                    Timestamp('1989-01-27 00:00:00'): 0.14211450517177582,
+                    Timestamp('1989-01-30 00:00:00'): 0.14398930966854095}}
+    test_df = pd.DataFrame(test_dict)
+    arima = ts.arima()
+    arima.fit(test_df, train_size=0.8, freq='B', q_vals=[0])
+    transformed_test_df = arima.transform(test_df)
+
+    for col in transformed_test_df.keys():
+        count_nans = arima.best_params[col].p + arima.best_params[col].d
+        assert(count_nans == transformed_test_df[col].isna().sum())
+
+    # Test (1,2,0) Model
+    diff_test_df_high = test_df['High'].diff().diff()
+    assert(transformed_test_df['High'][3] == (arima.best_params['High'].const + diff_test_df_high[2] * arima.best_params['High'].ar_coef[0]))
+    assert(transformed_test_df['High'][4] == (arima.best_params['High'].const + diff_test_df_high[3] * arima.best_params['High'].ar_coef[0]))
+    assert(transformed_test_df['High'][-1] == (arima.best_params['High'].const + diff_test_df_high[-2] * arima.best_params['High'].ar_coef[0]))
+
+    # Test (2,1,0) Model
+    diff_test_df_low = test_df['Low'].diff()
+    assert(isclose(transformed_test_df['Low'][3], (arima.best_params['Low'].const + diff_test_df_low[2] * arima.best_params['Low'].ar_coef[0] + diff_test_df_low[1] * arima.best_params['Low'].ar_coef[1]), abs_tol=1e-8))
+    assert(isclose(transformed_test_df['Low'][4], (arima.best_params['Low'].const + diff_test_df_low[3] * arima.best_params['Low'].ar_coef[0] + diff_test_df_low[2] * arima.best_params['Low'].ar_coef[1]), abs_tol=1e-8))
+    assert(isclose(transformed_test_df['Low'][-1], (arima.best_params['Low'].const + diff_test_df_low[-2] * arima.best_params['Low'].ar_coef[0] + diff_test_df_low[-3] * arima.best_params['Low'].ar_coef[1]), abs_tol=1e-8))
+
+    # Test (1,2,0) Model
+    diff_test_df_close = test_df['Close'].diff()[1:].diff()
+    first_day = pd.Series([np.nan])
+    first_day.index = [diff_test_df_close.index[0] - pd.DateOffset(days=1)]
+    first_day.name = 'Close'
+    diff_test_df_close = pd.concat([first_day, diff_test_df_close])
+    diff_test_df_close.index.name = "Date"
+
+    assert(transformed_test_df['Close'][4] == (arima.best_params['Close'].const + diff_test_df_close[3] * arima.best_params['Close'].ar_coef[0]))
+    assert(transformed_test_df['Close'][5] == (arima.best_params['Close'].const + diff_test_df_close[4] * arima.best_params['Close'].ar_coef[0]))
+    assert(transformed_test_df['Close'][-1] == (arima.best_params['Close'].const + diff_test_df_close[-2] * arima.best_params['Close'].ar_coef[0]))
+
+    print("All tests passed!")
```

Comment:
pls remove print and take a look at how the other test files are set up structurally